scripts/skills/perks/perk_rf_kingfisher.nut

```diff
@@ -35,82 +36,92 @@ this.perk_rf_kingfisher <- ::inherit("scripts/skills/skill", {
 
 	function onTargetHit( _skill, _targetEntity, _bodyPart, _damageInflictedHitpoints, _damageInflictedArmor )
 	{
-		if (!_targetEntity.isAlive())
+		if (!_targetEntity.isAlive() || ::Math.rand(1, 100) > this.m.Chance || !this.getContainer().getActor().isPlacedOnMap() || !_targetEntity.isPlacedOnMap())
 			return;
 
-		if (::Math.rand(1, 100) <= this.m.Chance)
+		local throwNetSkill = this.getContainer().getSkillByID("actives.throw_net");
+		if (!throwNetSkill.onVerifyTarget(this.getContainer().getActor().getTile(), _targetEntity.getTile()))
+			return;
+
+		local netItemScript = ::IO.scriptFilenameByHash(this.getContainer().getActor().getOffhandItem().ClassNameHash);
+
+		throwNetSkill.useForFree(_targetEntity.getTile());
+
+		local replacementNet = ::new(netItemScript)
+
+		this.getContainer().getActor().getItems().equip(replacementNet); // the original net is unequipped during use of throw net skill, so we equip a "new" net of the same type
+		foreach (skill in replacementNet.m.SkillPtrs)
```

Comment:
Before continuing here you need to double check whether the target is even netted.
In Vanilla it is possible to throw nets on enemies which are immune to the net effect.

Reply:
I see. I was not aware of this. I will add this check. Nevertheless, should we also make the net skill unusable on such targets?

Reply:
Yes, I suggest we add a `if (!targetEntity.getCurrentProperties().IsImmuneToRoot)` to the vanilla `throw_net` skill `onVerifyTarget` function.

That should fix this issue for the foreseeable future.